prow/githuboauth/githuboauth.go

```diff
@@ -54,13 +54,30 @@ type GitHubClientGetter interface {
 
 // OAuthClient is an interface for a GitHub OAuth client.
 type OAuthClient interface {
+	GetConfig() *oauth2.Config
```

Comment:
If the intent is to clone the client with a new final redirect URL, could we instead expose:

```
WithFinalRedrect(url *url.URL) OAuthClient
```
?

Reply:
That would keep us from leaking the internal config of the clients to users